### ISSUE TITLE
Dispatcher does not retry after network error

### ISSUE THREAD

**Issue by Roconda** [O2MC/dimml-mobile-sdk#4]
The dispatcher does not automatically retry sending data to an end point which results in an HTTP failure.

For instance
```
2018-06-22 13:33:42.264194+0200 app-obj-c[69591:4850218] lenght (570) Funnel -> ( {
  "application" : {
    "osVersion" : "11.4",
    "AppId" : "app-obj-c",
    "os" : "iOS",
    "connection" : "3G",
    "ip" : "192.168.x.x",
    "device" : "Simulator"
  },
  "tracked" : {
    "alias" : [
      {
        "alias" : "eventName",
        "event" : "alias",
        "identitiy" : "",
        "time" : "2018-06-22T13:33:38+02:00"
      }
    ],
    "eventName" : [
      {
        "alias" : "39C1FEBD-D7BE-430C-89EC-93610850C86C",
        "event" : "eventName",
        "identitiy" : "",
        "time" : "2018-06-22T13:33:37+02:00"
      }
    ]
  }
} ) has been dispatched to: http://127.0.0.1:3655/events
2018-06-22 13:33:42.272314+0200 app-obj-c[69591:4855701] TIC TCP Conn Failed [2:0x60000016f840]: 1:61 Err(61)
2018-06-22 13:33:42.272911+0200 app-obj-c[69591:4855701] Task <935D156C-4AC7-455C-9765-5B5E6A36AE4A>.<0> HTTP load failed (error code: -1004 [1:61])
2018-06-22 13:33:42.273039+0200 app-obj-c[69591:4859820] NSURLConnection finished with error - code -1004
```

The expected behaviour would be that the SDK would retry sending the data shortly in case of a failure.

**Comment by EdwinVanRooij** [comment]
Error could not be reproduced for Android. 
Test in branch `fix/dispatcher-network-retry`.

**Comment by EdwinVanRooij** [comment]
Fixed on Android as of #27. To be verified by @Roconda.

**Comment by Roconda** [comment]
The iOS implementation doesn't handle network errors correctly. It always assumes the http request has been successful and purges the funnel's queue.

It never reaches the else condition, see:

https://github.com/O2MC/dimml-mobile-sdk/blob/5518db54ff01a20d44c3a7e256b20a598a7b8a51/ios/sdk/O2MC/Dispatcher.m#L62-L68
